File: frontend/src/mocks/mockData.ts
import type { Store, Product, CapabilityType } from '../types/api';

// 1. Realistic stores in Istanbul (shared database reference)
export const mockStores: Store[] = [
  {
    id: 1,
    name: 'Turkcell Kadikoy TIM',
    address: 'Sogutlucesme Cd. No: 42, Kadikoy',
    city: 'Istanbul',
    district: 'Kadikoy',
    latitude: 40.9901,
    longitude: 29.0253,
    type: 'TIM',
    phone: '+90 216 555 0101',
    workingHours: '09:00 - 21:00'
  },
  {
    id: 2,
    name: 'Turkcell Besiktas TIM',
    address: 'Barbaros Blv. No: 12, Besiktas',
    city: 'Istanbul',
    district: 'Besiktas',
    latitude: 41.0428,
    longitude: 29.0075,
    type: 'TIM',
    phone: '+90 212 555 0102',
    workingHours: '09:00 - 21:00'
  },
  {
    id: 3,
    name: 'Turkcell Sisli TIM',
    address: 'Halaskargazi Cd. No: 150, Sisli',
    city: 'Istanbul',
    district: 'Sisli',
    latitude: 41.0602,
    longitude: 28.9877,
    type: 'TIM',
    phone: '+90 212 555 0103',
    workingHours: '09:00 - 22:00'
  },
  {
    id: 4,
    name: 'Turkcell Uskudar TIM',
    address: 'Hakimiyeti Milliye Cd. No: 80, Uskudar',
    city: 'Istanbul',
    district: 'Uskudar',
    latitude: 41.0267,
    longitude: 29.0152,
    type: 'TIM',
    phone: '+90 216 555 0104',
    workingHours: '09:00 - 20:00'
  },
  {
    id: 5,
    name: 'Turkcell Fatih TIM',
    address: 'Fevzipasa Cd. No: 210, Fatih',
    city: 'Istanbul',
    district: 'Fatih',
    latitude: 41.0186,
    longitude: 28.9497,
    type: 'TIM',
    phone: '+90 212 555 0105',
    workingHours: '09:00 - 20:00'
  },
  {
    id: 6,
    name: 'Turkcell Beyoglu TIM',
    address: 'Istiklal Cd. No: 75, Beyoglu',
    city: 'Istanbul',
    district: 'Beyoglu',
    latitude: 41.0370,
    longitude: 28.9764,
    type: 'TIM',
    phone: '+90 212 555 0106',
    workingHours: '10:00 - 22:00'
  },
  {
    id: 7,
    name: 'Turkcell Kadikoy Franchise 1',
    address: 'Moda Cd. No: 18, Kadikoy',
    city: 'Istanbul',
    district: 'Kadikoy',
    latitude: 40.9880,
    longitude: 29.0300,
    type: 'FRANCHISE',
    phone: '+90 216 555 0107',
    workingHours: '09:00 - 20:00'
  },
  {
    id: 8,
    name: 'Turkcell Besiktas Franchise 1',
    address: 'Sinanpasa Pasaji No: 5, Besiktas',
    city: 'Istanbul',
    district: 'Besiktas',
    latitude: 41.0410,
    longitude: 29.0090,
    type: 'FRANCHISE',
    phone: '+90 212 555 0108',
    workingHours: '09:00 - 20:00'
  },
  {
    id: 9,
    name: 'Turkcell Sisli Franchise 1',
    address: 'Abdi Ipekci Cd. No: 45, Nisantasi',
    city: 'Istanbul',
    district: 'Sisli',
    latitude: 41.0580,
    longitude: 28.9850,
    type: 'FRANCHISE',
    phone: '+90 212 555 0109',
    workingHours: '10:00 - 20:00'
  },
  {
    id: 10,
    name: 'Turkcell Uskudar Franchise 1',
    address: 'Baglarbasi Cd. No: 120, Uskudar',
    city: 'Istanbul',
    district: 'Uskudar',
    latitude: 41.0250,
    longitude: 29.0120,
    type: 'FRANCHISE',
    phone: '+90 216 555 0110',
    workingHours: '09:00 - 20:00'
  },
  {
    id: 11,
    name: 'Turkcell Fatih Franchise 1',
    address: 'Vatan Cd. No: 33, Fatih',
    city: 'Istanbul',
    district: 'Fatih',
    latitude: 41.0150,
    longitude: 28.9450,
    type: 'FRANCHISE',
    phone: '+90 212 555 0111',
    workingHours: '09:00 - 19:00'
  },
  {
    id: 12,
    name: 'Turkcell Kadikoy Franchise 2',
    address: 'Acibadem Cd. No: 88, Kadikoy',
    city: 'Istanbul',
    district: 'Kadikoy',
    latitude: 40.9850,
    longitude: 29.0200,
    type: 'FRANCHISE',
    phone: '+90 216 555 0112',
    workingHours: '09:00 - 20:00'
  },
  {
    id: 13,
    name: 'Turkcell Besiktas Franchise 2',
    address: 'Ortakoy Meydan No: 3, Besiktas',
    city: 'Istanbul',
    district: 'Besiktas',
    latitude: 41.0450,
    longitude: 29.0020,
    type: 'FRANCHISE',
    phone: '+90 212 555 0113',
    workingHours: '10:00 - 21:00'
  },
  {
    id: 14,
    name: 'Turkcell Sisli Franchise 2',
    address: 'Mecidiyekoy Yolu No: 12, Sisli',
    city: 'Istanbul',
    district: 'Sisli',
    latitude: 41.0620,
    longitude: 28.9920,
    type: 'FRANCHISE',
    phone: '+90 212 555 0114',
    workingHours: '09:00 - 20:00'
  },
  {
    id: 15,
    name: 'Turkcell Uskudar Franchise 2',
    address: 'Libadiye Cd. No: 200, Uskudar',
    city: 'Istanbul',
    district: 'Uskudar',
    latitude: 41.0290,
    longitude: 29.0200,
    type: 'FRANCHISE',
    phone: '+90 216 555 0115',
    workingHours: '09:00 - 20:00'
  }
];

// 2. Mock Products (Pasaj catalog)
export const mockProducts: Product[] = [
  { id: 1, name: 'iPhone 15 128GB', sku: 'APL-IPH15-128', category: 'Akıllı Telefonlar' },
  { id: 2, name: 'iPhone 15 Pro 256GB', sku: 'APL-IPH15P-256', category: 'Akıllı Telefonlar' },
  { id: 3, name: 'Samsung Galaxy S24 Ultra', sku: 'SAM-S24U-512', category: 'Akıllı Telefonlar' },
  { id: 4, name: 'Samsung Galaxy A55 128GB', sku: 'SAM-A55-128', category: 'Akıllı Telefonlar' },
  { id: 5, name: 'AirPods Pro Gen 2', sku: 'APL-APP2', category: 'Aksesuarlar' },
  { id: 6, name: 'Apple Watch Series 9', sku: 'APL-AW9-45', category: 'Akıllı Saatler' },
  { id: 7, name: 'Samsung Galaxy Watch 6', sku: 'SAM-GW6-44', category: 'Akıllı Saatler' },
  { id: 8, name: 'Xiaomi Redmi Note 13', sku: 'XIA-RN13-256', category: 'Akıllı Telefonlar' },
  { id: 9, name: 'Turkcell Superbox Router', sku: 'TKC-SBOX-LTE', category: 'Ağ ve İnternet' },
  { id: 10, name: 'Anker PowerCore 20k', sku: 'ANK-PC20', category: 'Aksesuarlar' }
];

// 3. Mock Stock levels mapping: key: productId-storeId, value: quantity
export const mockStocks: Record<string, number> = {
  // iPhone 15 stock (Product 1)
  '1-1': 10, // Kadikoy TIM - IN_STOCK
  '1-2': 4,  // Besiktas TIM - LOW
  '1-3': 0,  // Sisli TIM - OUT_OF_STOCK
  '1-4': 7,  // Uskudar TIM - IN_STOCK
  '1-5': 3,  // Fatih TIM - LOW
  '1-7': 12, // Kadikoy Franchise 1 - IN_STOCK
  '1-8': 2,  // Besiktas Franchise 1 - LOW
  '1-12': 0, // Kadikoy Franchise 2 - OUT_OF_STOCK

  // iPhone 15 Pro stock (Product 2)
  '2-1': 6,
  '2-2': 1,
  '2-3': 8,
  '2-6': 3,
  '2-9': 0,

  // Galaxy S24 Ultra (Product 3)
  '3-1': 3,
  '3-2': 12,
  '3-3': 0,
  '3-5': 5,
  '3-10': 2,

  // AirPods Pro (Product 5)
  '5-1': 20,
  '5-2': 15,
  '5-3': 18,
  '5-4': 25,
  '5-5': 8,
  '5-6': 30,
  '5-7': 5,
  '5-8': 2,
  '5-9': 0,
  '5-10': 12
};

// 4. Mock Capabilities mapping: key: storeId, value: list of capabilities
export const mockStoreCapabilities: Record<number, CapabilityType[]> = {
  1: ['NEW_LINE', 'DEVICE_DELIVERY', 'DEVICE_REPAIR', 'NUMBER_PORT', 'BILL_PAYMENT'], // TIMs support all
  2: ['NEW_LINE', 'DEVICE_DELIVERY', 'DEVICE_REPAIR', 'NUMBER_PORT', 'BILL_PAYMENT'],
  3: ['NEW_LINE', 'DEVICE_DELIVERY', 'DEVICE_REPAIR', 'NUMBER_PORT', 'BILL_PAYMENT'],
  4: ['NEW_LINE', 'DEVICE_DELIVERY', 'DEVICE_REPAIR', 'NUMBER_PORT', 'BILL_PAYMENT'],
  5: ['NEW_LINE', 'DEVICE_DELIVERY', 'DEVICE_REPAIR', 'NUMBER_PORT', 'BILL_PAYMENT'],
  6: ['NEW_LINE', 'DEVICE_DELIVERY', 'DEVICE_REPAIR', 'NUMBER_PORT', 'BILL_PAYMENT'],
  // Franchises support subsets
  7: ['NEW_LINE', 'NUMBER_PORT', 'BILL_PAYMENT'],
  8: ['NEW_LINE', 'DEVICE_DELIVERY', 'BILL_PAYMENT'],
  9: ['NEW_LINE', 'DEVICE_DELIVERY', 'NUMBER_PORT'],
  10: ['NEW_LINE', 'BILL_PAYMENT'],
  11: ['NEW_LINE', 'NUMBER_PORT', 'BILL_PAYMENT'],
  12: ['NEW_LINE', 'DEVICE_DELIVERY', 'BILL_PAYMENT'],
  13: ['NEW_LINE', 'NUMBER_PORT'],
  14: ['NEW_LINE', 'DEVICE_REPAIR', 'BILL_PAYMENT'],
  15: ['NEW_LINE', 'BILL_PAYMENT']
};

File: mobile/pasaj/Networking/MockData.swift
//
//  MockData.swift
//  pasaj
//
//  API contract'taki tiplere uygun sahte veri — backend hazır olmadan
//  UI geliştirmeye devam etmek için.
//

import Foundation

extension StoreWithDistance {
    init(store: Store, distance: Double, stockLevel: StockLevel?) {
        self.init(
            id: store.id,
            name: store.name,
            address: store.address,
            city: store.city,
            district: store.district,
            latitude: store.latitude,
            longitude: store.longitude,
            type: store.type,
            phone: store.phone,
            workingHours: store.workingHours,
            distance: distance,
            stockLevel: stockLevel
        )
    }
}

enum MockData {
    static let products: [Product] = [
        Product(id: 1, name: "iPhone 15 128GB", sku: "IP15-128-BLK", category: "Telefon"),
        Product(id: 2, name: "iPhone 15 Pro 256GB", sku: "IP15P-256-TIT", category: "Telefon"),
        Product(id: 3, name: "Samsung Galaxy S24 128GB", sku: "SGS24-128-BLK", category: "Telefon"),
        Product(id: 4, name: "Samsung Galaxy A55 128GB", sku: "SGA55-128-BLU", category: "Telefon"),
        Product(id: 5, name: "AirPods Pro 2", sku: "APP2-WHT", category: "Aksesuar"),
        Product(id: 6, name: "iPad 10.9\" Wi-Fi 64GB", sku: "IPAD-64-SLV", category: "Tablet"),
        Product(id: 7, name: "Xiaomi Redmi Note 13", sku: "XRN13-128-GRN", category: "Telefon"),
        Product(id: 8, name: "Apple Watch SE", sku: "AWSE-40-BLK", category: "Aksesuar"),
    ]

    static let stores: [Store] = [
        Store(id: 1, name: "Turkcell Kadıköy TİM", address: "Bahariye Cd. No:12", city: "İstanbul", district: "Kadıköy", latitude: 40.9838, longitude: 29.0297, type: .TIM, phone: "0216 555 01 01", workingHours: "09:00-20:00"),
        Store(id: 2, name: "Turkcell Beşiktaş TİM", address: "Barbaros Blv. No:45", city: "İstanbul", district: "Beşiktaş", latitude: 41.0422, longitude: 29.0067, type: .TIM, phone: "0212 555 02 02", workingHours: "09:00-20:00"),
        Store(id: 3, name: "Turkcell Üsküdar Bayi", address: "Hakimiyeti Milliye Cd. No:8", city: "İstanbul", district: "Üsküdar", latitude: 41.0234, longitude: 29.0151, type: .FRANCHISE, phone: "0216 555 03 03", workingHours: "10:00-19:00"),
        Store(id: 4, name: "Turkcell Şişli TİM", address: "Halaskargazi Cd. No:100", city: "İstanbul", district: "Şişli", latitude: 41.0602, longitude: 28.9877, type: .TIM, phone: "0212 555 04 04", workingHours: "09:00-20:00"),
        Store(id: 5, name: "Turkcell Maltepe Bayi", address: "Bağdat Cd. No:210", city: "İstanbul", district: "Maltepe", latitude: 40.9354, longitude: 29.1306, type: .FRANCHISE, phone: "0216 555 05 05", workingHours: "10:00-19:00"),
        Store(id: 6, name: "Turkcell Bakırköy TİM", address: "İstasyon Cd. No:33", city: "İstanbul", district: "Bakırköy", latitude: 40.9819, longitude: 28.8772, type: .TIM, phone: "0212 555 06 06", workingHours: "09:00-20:00"),
        Store(id: 7, name: "Turkcell Ataşehir Bayi", address: "Atatürk Mah. No:15", city: "İstanbul", district: "Ataşehir", latitude: 40.9923, longitude: 29.1244, type: .FRANCHISE, phone: "0216 555 07 07", workingHours: "10:00-19:00"),
        Store(id: 8, name: "Turkcell Beyoğlu TİM", address: "İstiklal Cd. No:150", city: "İstanbul", district: "Beyoğlu", latitude: 41.0361, longitude: 28.9770, type: .TIM, phone: "0212 555 08 08", workingHours: "09:00-21:00"),
    ]

    static let capabilityTypeOptions: [CapabilityTypeOption] = [
        CapabilityTypeOption(key: .NEW_LINE, label: "New Line Application"),
        CapabilityTypeOption(key: .DEVICE_DELIVERY, label: "Device Delivery"),
        CapabilityTypeOption(key: .DEVICE_REPAIR, label: "Device Repair / Technical Service"),
        CapabilityTypeOption(key: .NUMBER_PORT, label: "Number Portability"),
        CapabilityTypeOption(key: .BILL_PAYMENT, label: "Bill Payment"),
    ]

    // İşlemler sekmesi (capability sorgusu) için sabit sonuç. stockLevel'ın hepsi bilerek nil:
    // api-contract.md'de StoreCapabilityResult'ta bu alan yok, stok kavramı sadece Pasaj'a özgü.
    static let storesWithDistance: [StoreWithDistance] = [
        StoreWithDistance(store: stores[0], distance: 1.2, stockLevel: nil),
        StoreWithDistance(store: stores[1], distance: 2.8, stockLevel: nil),
        StoreWithDistance(store: stores[2], distance: 3.5, stockLevel: nil),
        StoreWithDistance(store: stores[3], distance: 4.1, stockLevel: nil),
        StoreWithDistance(store: stores[4], distance: 5.6, stockLevel: nil),
        StoreWithDistance(store: stores[5], distance: 6.3, stockLevel: nil),
    ].sorted { $0.distance < $1.distance }

    // Pasaj'ın ilk açılış görünümü: ürün/işlem bağımsız, sadece bayi konumlarını gösterir.
    static let allStoresOverview: [StoreWithDistance] = stores.enumerated().map { index, store in
        StoreWithDistance(store: store, distance: Double(index) * 0.7 + 0.5, stockLevel: nil)
    }.sorted { $0.distance < $1.distance }

    // Pasaj sekmesi: her ürün için farklı bir bayi/stok görünümü üretir (deterministik,
    // gerçek Haversine hesabı değil — sadece mock demo amaçlı).
    static func storesWithDistance(forProductId productId: Int) -> [StoreWithDistance] {
        let levels: [StockLevel] = [.IN_STOCK, .LOW, .OUT_OF_STOCK]
        return stores.enumerated().compactMap { index, store in
            guard (productId + index) % 4 != 0 else { return nil }
            let level = levels[(productId + index) % levels.count]
            let distance = (Double((productId * 3 + index * 7) % 40) / 10) + 0.8
            return StoreWithDistance(store: store, distance: (distance * 10).rounded() / 10, stockLevel: level)
        }
        .sorted { $0.distance < $1.distance }
    }
}
